data-search-widget/data-search-widget.js

```diff
@@ -1311,19 +1362,40 @@
           return;
         }
         var latlong = item.latitude + ',' + item.longitude;
-          if (!markers[latlong]) {
-              // add marker to map
-              markers[latlong] = L.marker(new L.LatLng(item.latitude, item.longitude));
-              markers[latlong].bindPopup(item.outletName);
-              markerClusters.addLayer(markers[latlong]);
-          }
+        if (!markers[latlong]) {
+          // add marker to map
+          markers[latlong] = L.marker(new L.LatLng(item.latitude, item.longitude));
+          markers[latlong].bindPopup(globalConfig.resultTemplate.markerPopupText(item));
+          markerClusters.addLayer(markers[latlong]);
+        }
       }
     });
-    //map.addLayer(markerClusters);
-    markerClusters.addTo(map);
+    // Update map bounds
+    if (lastFilteredResults != mapsData) {
```

Comment:
```suggestion
    if (lastFilteredResults != mapsData || updateMapBounds) {
```
Logic using `updateMapBounds` argument to control whether map bounds should be changed.